src/charm.py

```diff
@@ -291,6 +310,31 @@ def create_kubeconfigs(self):
             user="kube-proxy",
         )
 
+    def configure_observability(self):
+        """Apply observability configurations to the cluster."""
+        # Apply Clusterrole and Clusterrole binding for COS observability
+        if self.unit.is_leader():
+            kubectl("apply", "-f", "templates/observability.yaml")
+        # Issue a token for metrics scraping
+        cos_user = f"system:cos:{kubernetes_snaps.get_node_name()}"
```

Comment:
`get_node_name` needs to be called with fqdn=True in case of aws.  Rebase and look at other examples of this call.  Consider adding a `self.get_node_name()` do this class so we don't make this mistake